src/Core/src/Handlers/View/AndroidBatchPropertyMapper.cs

```diff
@@ -1,56 +1,51 @@
-﻿using System;
+﻿#if ANDROID
+
+using System;
 using System.Collections.Generic;
 
-namespace Microsoft.Maui.Handlers
-{
-#if ANDROID
-	class AndroidBatchPropertyMapper<TVirtualView, TViewHandler> : PropertyMapper<TVirtualView, TViewHandler>
-		where TVirtualView : IElement
-		where TViewHandler : IElementHandler
-	{
-		// During mass property updates, this list of properties will be skipped
-		public static HashSet<string> SkipList = new(StringComparer.Ordinal)
-		{ 
-			nameof(IView.Visibility),
-			nameof(IView.MinimumHeight),
-			nameof(IView.MinimumWidth),
-			nameof(IView.IsEnabled),
-			nameof(IView.Opacity),
-			nameof(IView.TranslationX),
-			nameof(IView.TranslationY),
-			nameof(IView.Scale),
-			nameof(IView.ScaleX),
-			nameof(IView.ScaleY),
-			nameof(IView.Rotation),
-			nameof(IView.RotationX),
-			nameof(IView.RotationY),
-			nameof(IView.AnchorX),
-			nameof(IView.AnchorY),
-		};
+namespace Microsoft.Maui.Handlers;
 
-		public AndroidBatchPropertyMapper(params IPropertyMapper[] chained) : base(chained) { }
+file class AndroidPropertyMapperInfo
+{
+	public static readonly ISet<string> UpdatePropertiesSkipList = new HashSet<string>(StringComparer.Ordinal)
+	{ 
+		nameof(IView.Visibility),
+		nameof(IView.MinimumHeight),
+		nameof(IView.MinimumWidth),
+		nameof(IView.IsEnabled),
+		nameof(IView.Opacity),
+		nameof(IView.TranslationX),
+		nameof(IView.TranslationY),
+		nameof(IView.Scale),
+		nameof(IView.ScaleX),
+		nameof(IView.ScaleY),
+		nameof(IView.Rotation),
+		nameof(IView.RotationX),
+		nameof(IView.RotationY),
+		nameof(IView.AnchorX),
+		nameof(IView.AnchorY),
+	};
+}
 
-		public override IEnumerable<string> GetKeys()
+class AndroidBatchPropertyMapper<TVirtualView, TViewHandler>(params IPropertyMapper[] chained)
+	: PropertyMapper<TVirtualView, TViewHandler>(chained)
+	where TVirtualView : IElement
+	where TViewHandler : IElementHandler
+{
+	private protected override void UpdatePropertiesCore(IElementHandler viewHandler, IElement virtualView)
+	{
+		foreach (var mapper in MergedMappers)
 		{
-			foreach (var key in _mapper.Keys)
+			// When doing the initial properties batch updates, ignore properties in the skip list.
+			// These will be handled by ViewHandler.SetVirtualView() instead.
```

Comment:
For reference: https://github.com/dotnet/maui/pull/3372/files#diff-928414507a5c749dc42759124e98c6bf08bdb6b7dff26549616bfc683a65edec